azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -462,11 +462,8 @@ protected LogData getFlowLogData(final ExecutableFlow exFlow, final int offset,
     } else {
       LogData logData = this.nearlineExecutionLogsLoader.fetchLogs(exFlow.getExecutionId(), "", 0,
           offset, length, exFlow.getEndTime());
-      // Return offline logs if nearline logs are empty or the flow in kubernetes pod crashed
-      if (offlineLogsLoaderEnabled && offlineExecutionLogsLoader.isPresent() &&
-          (logData == null ||
-              (exFlow.getDispatchMethod() == DispatchMethod.CONTAINERIZED &&
-                  (exFlow.getStatus() == Status.KILLED || exFlow.getStatus() == Status.EXECUTION_STOPPED)))) {
+      // Return offline logs if nearline logs are empty
+      if (offlineLogsLoaderEnabled && offlineExecutionLogsLoader.isPresent() && logData == null) {
```

Comment:
Since this logic is duplicated below would it make sense to refactor into a methd?

Reply:
I feel like it is okay to duplicate here. One condition is 
`offlineLogsLoaderEnabled && offlineExecutionLogsLoader.isPresent() && logData == null`
Another is 
`!nearlineOnly && offlineLogsLoaderEnabled && offlineExecutionLogsLoader.isPresent() && logData == null` (so slightly different).

If we want to refactor the last three ANDs into a method, the method name itself is hard to figure out and it only combines three AND operations which seems to be an overkill.

Reply:
This makes sense, thank you